src/index.ts

```diff
@@ -36,11 +29,19 @@ export const eventHandler = async (
   settings: ComponentSettings
 ) => {
   const { client, payload } = event
-  const neededFetch = []
 
-  const conversionId = payload.conversionId || settings.conversionId
-  delete payload.conversionId
+  // set the _gcl_aw cookie if _gl or gclid query params exists
+  setGclAwCookie(client)
+
+  // if pageview, run conversion linker if enabled and return
+  if (eventType === 'pageview') {
+    if (settings.domains) {
+      conversionLinkerHandler(client, settings)
+    }
+    return
```

Comment:
why do we return here? if it's a pageview, we eventually run conversionLinkerHandler but not the rest of the eventHandler?

Reply:
Yes, this is it. See Yo'av's comment:

```
sounds to me like a good flow would be:

pageview:
- save the cookie
- do the links rewriting (if enabled)

conversion:
- save the cookie
- build the request and send it

remarketing:
- save the cookie
- build the request and send it

so saving of the cookie should be a separate on-off function that's called for all three, and the second step is different between pageview / conversion & remarktgin
```